TipCatDotNet.Api/Services/HospitalityFacilities/MemberService.cs

```diff
@@ -288,7 +289,10 @@ Task<Result<int>> AddMember()
         private async Task<Result<int>> AssignMemberCode(int memberId, CancellationToken cancellationToken)
         {
             var memberCode = MemberCodeGenerator.Compute(memberId);
-            var qrCodeUrl = $"/{memberCode}"; // TODO: add real code generation and url here
+            var qrCodeUrl = await _qrCodeGenerator
+                .Generate(memberCode, cancellationToken)
+                .Ensure(url => !string.IsNullOrWhiteSpace(url), "Url wasn't created by amazon's endpoint!")
+                .Finally(result => result.IsSuccess ? result.Value : result.Error);
```

Comment:
Let's rewrite this part. As of now you convert both positive and negative results to a `qrCodeUrl`. In other words, an error may be placed in a DB without any tracking. What we could do to avoid such behavior? Either throw an exception, either leave the field empty.
```csharp
var (_, isFailure, qrCodeUrl, error) = await _qrCodeGenerator
    .Generate(memberCode, cancellationToken);

if (isFailure)
{
    _logger.LogAmazonS3Unreachable(error);
    qrCodeUrl = string.Empty;
}
...
member.QrCodeUrl = qrCodeUrl;
```
and make a method to regenerate the code later, or
```csharp
var (_, isFailure, qrCodeUrl, error) = await _qrCodeGenerator
    .Generate(memberCode, cancellationToken);

if (isFailure)
{
   _logger.LogAmazonS3Unreachable(error);
    throw new Exception(error);
}
```

Reply:
Let's make the first option, because we'll need regenerations in the near future